ros_gz_bridge/test/utils/ros_test_msg.cpp

```diff
@@ -956,6 +959,12 @@ void compareTestMsg(const std::shared_ptr<sensor_msgs::msg::Imu> & _msg)
   compareTestMsg(_msg->orientation);
   compareTestMsg(std::make_shared<geometry_msgs::msg::Vector3>(_msg->angular_velocity));
   compareTestMsg(std::make_shared<geometry_msgs::msg::Vector3>(_msg->linear_acceleration));
+
+  for (int i = 1; i < 10; ++i){
```

Comment:
why not 0 to 9 ?

Reply:
Nice catch, it was because in the `createTestMsg` method above, the expected covariance arrays are {1,2,3....,9}.
But I can instead loop from 0 to 9 and do 
```cpp
EXPECT_EQ(_msg->...covariance[i] = i+1);
...
```
By the way, it is weird that the unit test passed. I'll double check

Reply:
So I double checked, the unit test were indeed not passing. I fixed them, as explained above.
But other tests were failing as well ( NavSat & GpsFix), so I went ahead and fixed them as well.